Examples/UIExplorer/AsyncStorageExample.js

```diff
@@ -29,15 +29,16 @@ var COLORS = ['red', 'orange', 'yellow', 'green', 'blue'];
 
 var BasicStorageExample = React.createClass({
   componentDidMount() {
-    AsyncStorage.getItem(STORAGE_KEY, (error, value) => {
-      if (error) {
-        this._appendMessage('AsyncStorage error: ' + error.message);
-      } else if (value !== null) {
-        this.setState({selectedValue: value});
-        this._appendMessage('Recovered selection from disk: ' + value);
-      } else {
-        this._appendMessage('Initialized with no selection on disk.');
-      }
+    AsyncStorage.getItem(STORAGE_KEY)
+      .then(value => {
+        if(value !== null){
+          this.setState({selectedValue: value});
+          this._appendMessage('Recovered selection from disk: ' + value);
+        } else {
+          this._appendMessage('Initialized with no selection on disk.');
+        }
+      })
+      .catch((error) => this._appendMessage('AsyncStorage error: ' + error.message));
```

Comment:
I ran some tests and it seems to work fine without calling `done()`:
```
new Promise((res, rej) => { throw new Error("SimulatedException"); })
.then((msg)  => console.log('uncaught: ' + msg))
.catch((msg) => console.warn('caught: ' + msg)); // outputs caught: SimulatedException

new Promise((res, rej) => { rej("REJECTED"); })
.then((msg)  => console.log('uncaught: ' + msg))
.catch((msg) => console.warn('caught: ' + msg)); // outputs caught: REJECTED
```
That being said, I'm happy to add the `done()` at the end of the promise chain if you'd prefer it be in there.